src/metrics/peerstore.go

```diff
@@ -359,6 +361,41 @@ func (c *PeerStore) ExportToCSV(filePath string) error {
 	return nil
 }
 
+// Function that save the raw Data of all the peers to a db (Using BoltDB)
+// Using the peerID as a key, and the Peer struct marshalled into JSON as a value
+func (c *PeerStore) PersistToDb() {
+	db, err := bolt.Open("../../peerstore.db", 0600, &bolt.Options{Timeout: 5 * time.Second})
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer db.Close()
+
+	err = db.Update(func(tx *bolt.Tx) error {
+		b, err := tx.CreateBucketIfNotExists([]byte("PeerStore"))
+		if err != nil {
+			return err
+		}
+
+		c.PeerStore.Range(func(k, val interface{}) bool {
+			peerData := val.(Peer)
+			buf, err := json.Marshal(peerData)
+			if err != nil{
+				log.Fatal(err)
+				return false
+			}
+			b.Put([]byte(peerData.PeerId), buf)
+			return true
+		})
+
+		return nil
+	})
+
+	if err != nil{
+		log.Fatal(err)
```

Comment:
Same, return the error so that it can be handled in the caller side. We don't want our crawlers to stop executing :)